source/MessageLogPanel.cpp

```diff
@@ -73,13 +73,29 @@ void MessageLogPanel::Draw()
 	Point pos = Screen::BottomLeft() + Point(PAD, scroll);
 	for(const auto &it : messages)
 	{
+		if(importantOnly && (it.second == Messages::Importance::Low || it.second == Messages::Importance::High))
+			continue;
+
 		messageLine.Wrap(it.first);
 		pos.Y() -= messageLine.Height();
 		if(pos.Y() >= Screen::Top() - 3 * font.Height())
 			messageLine.Draw(pos, *Messages::GetColor(it.second, true));
 	}
 
 	maxScroll = max(0., scroll - pos.Y() + Screen::Top());
+
+	// Draw the filter box.
+	const Sprite *filterBack = SpriteSet::Get("ui/message log key");
+	pos = Screen::BottomLeft() + Point(width + 40., 0.);
+	SpriteShader::Draw(filterBack, pos + Point(.5 * filterBack->Width(), -.5 * filterBack->Height()));
+	const Color color[2] = {*GameData::Colors().Get("medium"), *GameData::Colors().Get("bright")};
+	const Sprite *box[2] = {SpriteSet::Get("ui/unchecked"), SpriteSet::Get("ui/checked")};
+	pos += Point(30., -30.);
+	SpriteShader::Draw(box[importantOnly], pos);
+	Point off = Point(10., -.5 * font.Height());
+	font.Draw("Hide less _important", pos + off, color[importantOnly]);
+	font.Draw("messages", pos + off + Point(0., 20.), color[importantOnly]);
+	AddZone(Rectangle(pos + Point(64., 10.), Point(140., 40.)), [this](){ importantOnly = !importantOnly; });
```

Comment:
Can't most/all of this be done through an interface?

Reply:
I figured out that currently using an interface here would require _really_ hacky methods and I'm waiting for a response on 6424. I was thinking of adding extensive interface support to all conversation-like panels anyway, so I'll just leave this hardcoded for now.

Reply:
I don't think quyykk is coming back to that any time soon. Maybe you could take if over for him?

Reply:
> I don't think quyykk is coming back to that any time soon. Maybe you could take if over for him?

yeah sure